tests/unit/test_charm.py

```diff
@@ -42,36 +50,121 @@ def test_action_failures(self):
 
-        self.harness.update_config({"database": "bar"})
+    def test_get_unit_status(self):
+        self.harness.set_leader(True)
+        self.harness.update_config({"database-name": "foo"})
         self.harness.charm._on_config_changed(Mock())
-        self.assertEqual(self.harness.model.unit.status, ActiveStatus("database: bar"))
-        self.assertEqual(self.harness.charm.config["database"], "bar")
-        self.assertEqual(self.harness.charm.mysql.database, "bar")
-        self.assertEqual(self.harness.charm.postgresql.database, "bar")
-        self.assertEqual(self.harness.charm.mongodb.database, "bar")
+        self.assertEqual(
+            self.harness.model.unit.status,
+            BlockedStatus("Please relate the data-integrator with the desired product"),
+        )
+        self.assertEqual(self.harness.charm.config["database-name"], "foo")
+
+        self.rel_id = self.harness.add_relation("mysql", "mysql")
+        self.harness.add_relation_unit(self.rel_id, "mysql/0")
+
+        # Simulate sharing the credentials of a new created database.
+        self.harness.update_relation_data(
+            self.rel_id,
+            "mysql",
+            {
+                "username": "test-username",
+                "password": "test-password",
+            },
+        )
+
+        self.assertEqual(
+            self.harness.model.unit.status,
+            ActiveStatus(),
+        )
+
+        self.harness.update_config({"database-name": "foo1"})
+        self.harness.charm._on_config_changed(Mock())
+
+        self.assertEqual(
+            self.harness.model.unit.status,
+            BlockedStatus("To change database name, please remove relation and add it again"),
+        )
+
+        self.harness.remove_relation(self.rel_id)
+        self.harness.charm._on_config_changed(Mock())
+        self.assertEqual(
+            self.harness.model.unit.status,
+            BlockedStatus("Please relate the data-integrator with the desired product"),
+        )
+
+        self.harness.update_config({"topic-name": "bar"})
+        self.harness.charm._on_config_changed(Mock())
+        self.assertEqual(self.harness.charm.config["topic-name"], "bar")
+        self.assertEqual(
+            self.harness.model.unit.status,
+            BlockedStatus("Please relate the data-integrator with the desired product"),
+        )
+
+        self.rel_id = self.harness.add_relation("kafka", "kafka")
+        self.harness.add_relation_unit(self.rel_id, "kafka/0")
+
+        # Simulate sharing the credentials of a new created topic.
+        self.harness.update_relation_data(
+            self.rel_id,
+            "kafka",
+            {
+                "topic": "bar",
+                "username": "test-username",
+                "password": "test-password",
+            },
+        )
+        self.harness.charm._on_config_changed(Mock())
+        self.assertEqual(
+            self.harness.model.unit.status,
+            ActiveStatus(""),
+        )
```

Comment:
**issue:** For `ActiveStatus` we should not have a message, so here I would remove the `""`, but the `assertEquals` is good here.